src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -372,9 +372,9 @@ internal async Task IndexMethodAsyncCore(MethodInfo method, IFunctionIndexCollec
                 IsDisabled = disabled,
                 HasCancellationToken = hasCancellationToken,
                 TimeoutAttribute = TypeUtility.GetHierarchicalAttributeOrNull<TimeoutAttribute>(method) ?? defaultTimeout,
-                SingletonAttributes = method.GetCustomAttributes<SingletonAttribute>(),
-                MethodLevelFilters = method.GetCustomAttributes().OfType<IFunctionFilter>(),
-                ClassLevelFilters = method.DeclaringType.GetCustomAttributes().OfType<IFunctionFilter>()
+                SingletonAttributes = method.GetCustomAttributes<SingletonAttribute>().ToArray(),
+                MethodLevelFilters = method.GetCustomAttributes().OfType<IFunctionFilter>().ToArray(),
+                ClassLevelFilters = method.DeclaringType.GetCustomAttributes().OfType<IFunctionFilter>().ToArray()
```

Comment:
Is there a perf gain from enumerating these immediately?

Reply:
Likely not a big one, but just like we did already for TimeoutAttribute, since this metadata is static once indexed, I'm just setting it up front. Don't need to requery for it on each invocation.